staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go

```diff
@@ -664,58 +638,64 @@ func TestForbiddenFieldsInSchema(t *testing.T) {
 	}
 	defer tearDown()
 
-	noxuDefinitions := newNoxuValidationCRDs(apiextensionsv1beta1.NamespaceScoped)
+	noxuDefinitions := newNoxuValidationCRDs()
 	for i, noxuDefinition := range noxuDefinitions {
 		for _, v := range noxuDefinition.Spec.Versions {
 			// Re-define the CRD to make sure we start with a clean CRD
-			noxuDefinition := newNoxuValidationCRDs(apiextensionsv1beta1.NamespaceScoped)[i]
+			noxuDefinition := newNoxuValidationCRDs()[i]
 			validationSchema, err := getSchemaForVersion(noxuDefinition, v.Name)
 			if err != nil {
 				t.Fatal(err)
 			}
-			validationSchema.OpenAPIV3Schema.AdditionalProperties.Allows = false
-
-			_, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
-			if err == nil {
-				t.Fatalf("unexpected non-error: additionalProperties cannot be set to false")
-			}
-
-			validationSchema.OpenAPIV3Schema.Properties["zeta"] = apiextensionsv1beta1.JSONSchemaProps{
+			// v1 doesn't allow additional properties at all. this isn't necessary in v1.
```

Comment:
This needs to be put back to v1beta1 until 1.22, when it can be removed because it is testing client behavior, not etcd/controller behavior.

Reply:
we support it, but not at the root.

Reply:
as this to the schema instead:
```
validationSchema.OpenAPIV3Schema.Properties["eta"]: {
  type: object
  additionalProperties: false
}
```